src/LibraryUtilities.ts

```diff
@@ -563,11 +563,16 @@ export function removeEmptyNodes(items: ItemData[]) {
             if (removeEmptyNodes(item.childItems)) {
                 i--;
             }
-        }else if ( (item.itemType === "section" && item.contextData !== "Add-ons") || item.itemType === "category" || item.itemType === "group" || item.itemType === "coregroup") {
+        }
+		// Do not remove the Add-ons section from the list even when it has no child elements, as we want to show this section at all times.
+		else if (item.itemType === "section" && item.contextData === "Add-ons") {
```

Comment:
how is this different from your original changes?

Reply:
These changes are same, just added it to a different else block to be clear. As we are not removing the add-ons section here, had to do the filtering on the search side.